src/bandwidth: Drop `atomic` dependency

hrxi opened the pull request:
Replace `atomic::Atomic<u64>` by `std::sync::atomic:AtomicU64`.

The original motivation of using `atomic::Atomic<u64>` instead of
`std`'s version in https://github.com/libp2p/rust-libp2p/pull/1670 was the following:

> I used the atomic crate and an Atomic<u64> because the AtomicU64 type
> isn't available on all architectures. The atomic crate automatically
> falls back to using a Mutex on platforms that don't support AtomicU64.

This argumentation is moot because the crate directly depends on
`libp2p-core` which also uses `AtomicU64`.

thomaseizinger reviewed (changes requested):
Thanks for the contribution!

1. Can you write a few lines about this change? i.e. is the `atomic` crate fully superseded by the std lib?
2. We don't have a lockfile checked into the repository because we are only developing a library here. Can you post the output of `cargo tree -i -p atomic` to show that we are actually dropping the dependency?
3. Rustfmt doesn't seem to be happy with the changes, please fix that :)

hrxi committed:
Drop `atomic` dependency

Replace `atomic::Atomic<u64>` by `std::sync::atomic:AtomicU64`.

The original motivation of using `atomic::Atomic<u64>` instead of
`std`'s version in #1670 was the following:

> I used the atomic crate and an Atomic<u64> because the AtomicU64 type
> isn't available on all architectures. The atomic crate automatically
> falls back to using a Mutex on platforms that don't support AtomicU64.

This argumentation is moot because the crate directly depends on
`libp2p-core` which also uses `AtomicU64`.
